src/attachment/index.ts

```diff
@@ -0,0 +1,351 @@
+      }
+    }
+
+    private _handleDragOver(event: DragEvent) {
+      event.stopPropagation();
+      if (this._isFileOrDirectoryDrag(event)) {
+        event.preventDefault();
+      }
+    }
+
+    private _handleDragDrop(event: DragEvent) {
+      event.preventDefault();
+      this._handleDragLeave();
+      if (this._isFileDrop(event)) {
+        this._addFiles(event);
+      }
+    }
+
+    private _isFileDrop(event: DragEvent) {
+      // handle Chrome, Firefox, Edge, Safari
+      if (event.dataTransfer && event.dataTransfer.items) {
+        for (let i = 0; i < event.dataTransfer.items.length; i++) {
+          if (
+            typeof event.dataTransfer.items[i].webkitGetAsEntry ===
+              "function" &&
+            event.dataTransfer.items[i].webkitGetAsEntry()!.isDirectory
+          ) {
+            return false;
+          }
+        }
+      }
+      return true;
+    }
+
+    private _handleDragLeave() {
+      this._dragEnterCounter--;
+
+      if (this._dragEnterCounter === 0) {
+        this._groupFilesEl.style.height = "auto";
+        this._isDraging = false;
+      }
+    }
+
+    private _handleChangeFiles(event: Event) {
+      event.preventDefault();
+      event.stopPropagation();
+      this._addFiles(event);
+    }
+
+    private _addFiles(event: any) {
+      if (this.files) {
+        let addedFiles = event.dataTransfer
+          ? event.dataTransfer.files
+          : event.target.files;
+        addedFiles = Object.keys(addedFiles).map((e) => {
+          return addedFiles[e];
+        });
+        const tempFileList = [...this.files];
+        const fileIndex = addedFiles.map((_item: any, index: number) => {
+          return tempFileList.length + index;
+        });
+        addedFiles.forEach((addedFile: FileItem) => this.files.push(addedFile));
+        const detail = {
+          oldFiles: tempFileList,
+          files: this.files,
+          type: "add-file",
+          fileIndex: fileIndex,
+        };
+        dispatchCustomEvent(this, "change", detail);
+        this.requestUpdate();
+      }
+      this._inputEl.value = "";
+    }
+
+    private _getFileSize(size: string | number) {
+      if (typeof size === "number") {
+        return this._formatFileSize(size);
+      }
+      return validatePositiveInteger(size)
+        ? this._formatFileSize(parseInt(size, 10))
+        : ATTACHMENT.INVALID_SIZE_ERROR;
+    }
+
+    private _formatFileSize(size: number) {
+      if (size >= ATTACHMENT.UNIT_LENGTH.ONE_GB) {
+        return Math.round(size / ATTACHMENT.UNIT_LENGTH.ONE_GB) + " GB";
+      } else if (size >= ATTACHMENT.UNIT_LENGTH.ONE_MB) {
+        return Math.round(size / ATTACHMENT.UNIT_LENGTH.ONE_MB) + " MB";
+      } else if (size >= ATTACHMENT.UNIT_LENGTH.ONE_KB) {
+        return Math.round(size / ATTACHMENT.UNIT_LENGTH.ONE_KB) + " KB";
+      }
+      return Math.round(size) + " bytes";
+    }
+
+    private _isFileOrDirectoryDrag = (event: DragEvent) => {
```

Comment:
private _isFileOrDirectoryDrag = (event: DragEvent) => {
      if (!event.dataTransfer) return false;

      if (event.dataTransfer.items !== undefined) {
        for (let i = 0; i < event.dataTransfer.items.length; i++) {
          if (event.dataTransfer.items[i].kind.toLowerCase() === "file") {
            return true;
          }
        }
      }

      if (event.dataTransfer.types !== undefined) {
        for (let i = 0; i < event.dataTransfer.types.length; i++) {
          if (event.dataTransfer.types[i].toLowerCase() === "files") {
            return true;
          }
        }
      }
      return false;
    };